content/blog/a-summary-of-the-july-retrospectives/index.md

```diff
@@ -0,0 +1,78 @@
+
+ 
+## Respins
+We have covered the issues introduced in our repositories that caused minor delays, now let's look at changes in the upstream OpenJDK repositories that resulted in the need to respin every JDK version of this release, a costly activity. 
+
+For the JDK8u345 respin, the issue is described in [JDK-8194154](https://bugs.openjdk.org/browse/JDK-8194154), which describes the crash that would occur when changing the "user.dir" property.  While changing user.dir is discouraged, some applications still do it, notably Gradle.  This is an interesting dilemna, where the feature is highly discouraged, but still possible, and therefore it is used in the field by some number of applications.  This highlights the fact that we need to be testing with a broad set of external applications, especially ones that are used widely.  We are actively incorporating a set of tests in the AQAvit suite for this purpose, but need to progress this work.  
+
+JDK11.0.16.1, JDK17.0.4.1 and JDK18.0.2.1: c2 regression (excessive memory consumption after 17.0.4 update): [JDK-8292260](https://bugs.openjdk.org/browse/JDK-8292260) - This defect caused *intermittent* crashes and was introduced as part of a CVE patch.  Since our Eclipse Foundation project is not a member of the Vulnerability Group (VG), we do not have access to these patches ahead of the GA tags being available.  
+
+Looking at the last 2 years and set of unique issues forcing the need to respin (**Aug 2020:JDK-8249677**, **Nov 2020:JDK-8250861**, Feb 2022:JDK-8218546, Aug 2022:JDK-8290832, **Aug 2022:JDK-8292260**), the **majority** were introduced by CVE patches.  It would be good to understand what level of testing the VG does with these CVE patches ahead of releasing them into the OpenJDK code base, and assess what needs to be changed in order to identify these issues before delivery to avoid the churn of a respin.  
+
+In these cases, it was less likely that we would have been able to catch these failures, as we did not have weeks to soak these changes and see the intermittent issues.  The first time we have a chance to build and test with these CVE patches is the day the GA tags are made available which is Day 1 of our release period.  In order for us to have access to these patches earlier we would need to join the VG.  While there has been an effort for Eclipse Foundation to join the VG, Oracle has denied the request.  One could argue that the VG would benefit from an additional member that could assist in testing these patches more heavily and across a very broad range of platforms to boot.  In absence of membership, we will have to look at soak testing our release builds a bit longer, the unfortunate side-effect being that it could potentially increase the time it takes us to release products.    
```

Comment:
Could we phrase this as "... Oracle has so far denied the request because `<reasons>`" or something like that?

Reply:
I was going with the public minutes of the Adoptium Steering committee call from last week, which I do not think gives a reason.  I will see what can be added here as to their decision to date.